tests/unittest/test_sampler.py

```diff
@@ -95,3 +95,17 @@ def test_sorted_bucket_sampler(seq_lengths, mult, batch_size, shuffle):
     for batch_sample_ids in sampler:
         total_sampled_ids.extend(batch_sample_ids)
     assert len(set(total_sampled_ids)) == len(total_sampled_ids) == N
+
+
+@pytest.mark.parametrize('num_samples', 30)
+@pytest.mark.parametrize('num_parts', [3, 7])
+def test_split_sampler(num_samples, num_parts):
```

Comment:
Can you test that all indices are actually sampled? Ie. just add each sampled index to a `set` and make sure the length of that set equals `num_samples` as well as none of them is out of bounds?

Reply:
Updated test. Thanks for the suggestion!